dali/test/python/test_utils.py

```diff
@@ -435,3 +435,98 @@ def np_type_to_dali(type):
         np.float64: types.FLOAT64,
     }
     return np_types_to_dali_dict[type]
+
```

Comment:
Hmm, for know if callback raises anything else than the stopinteration it just crashes the worker process. Stacktrace with error of the workers gets printed to stdout so you would see that in the console. And parent process would kill all the other processes and raise error about workers unexpected exit. Can change it to reraise the error from worker in the parent process, though I am not entirely sure if that's really desirable. What do you think?

Reply:
The question is if I can launch python, do something, get error and again do something or I need to restart python?
I think the user should be able to catch error, handle it somehow and proceed (like create another pipeline).

Reply:
Okay, not really, like user can catch something like BrokenPipe but I believe it's not that great. 

I see two options here:
1. add explicitly some exception like WorkerError raised each time workers dies unexpectedly instead of raising BrokenError/EndOfFile etc. (it depends on what exactly the main process was doing when worker exited) or 
2. explicitly forward the exception to the main process, though here are two issues: 
2a. should it follow standard communication flow which implies some delay between raising an error in a worker and getting it in the main process (due to scheduling) or should we have dedicated hot line to pass the exception to the main process, 
2b. the other thing is that exception instance must be picklable - it should be most of the time, though it's technically possible for user to come up with some unpicklable custom exception.

Reply:
> 2a. should it follow standard communication flow which implies some delay between raising an error in a worker and getting it in the main process (due to scheduling) or should we have dedicated hot line to pass the exception to the main process,

I think that is fine, as the user cannot expect where the main process should be when and where the child one, besides the sync point at the end.